src/pages/home/report/ReactionList/PopoverReactionList.js

```diff
@@ -0,0 +1,218 @@
+        };
+
+        this.onPopoverHideActionCallback = () => {};
+        this.reactionListAnchor = undefined;
+        this.showReactionList = this.showReactionList.bind(this);
+
+        this.hideReactionList = this.hideReactionList.bind(this);
+        this.measureContent = this.measureContent.bind(this);
+        this.measureReactionListPosition = this.measureReactionListPosition.bind(this);
+        this.getReactionListMeasuredLocation = this.getReactionListMeasuredLocation.bind(this);
+
+        this.dimensionsEventListener = null;
+
+        this.contentRef = React.createRef();
+    }
+
+    componentDidMount() {
+        this.dimensionsEventListener = Dimensions.addEventListener('change', this.measureReactionListPosition);
+    }
+
+    shouldComponentUpdate(nextProps, nextState) {
+        const previousLocale = lodashGet(this.props, 'preferredLocale', 'en');
+        const nextLocale = lodashGet(nextProps, 'preferredLocale', 'en');
+        return this.state.isPopoverVisible !== nextState.isPopoverVisible
+            || this.state.popoverAnchorPosition !== nextState.popoverAnchorPosition
+            || previousLocale !== nextLocale;
+    }
+
+    componentWillUnmount() {
+        if (!this.dimensionsEventListener) {
+            return;
+        }
+        this.dimensionsEventListener.remove();
+    }
+
+    /**
+     * Get the PopoverReactionList anchor position
+     * We calculate the achor coordinates from measureInWindow async method
+     *
+     * @returns {Promise<Object>}
+     */
+    getReactionListMeasuredLocation() {
+        return new Promise((resolve) => {
+            if (this.reactionListAnchor) {
+                this.reactionListAnchor.measureInWindow((x, y) => resolve({x, y}));
+            } else {
+                resolve({x: 0, y: 0});
+            }
+        });
+    }
+
+    /**
+     * Show the ReactionList modal popover.
+     *
+     * @param {Object} [event] - A press event.
+     * @param {Element} reactionListAnchor - reactionListAnchor
+     * @param {Array} users - Array of personal detail objects
+     * @param {String} emojiName - Name of emoji
+     * @param {Array} emojiCodes - The emoji codes to display in the bubble.
+     * @param {Number} emojiCount - Count of emoji
+     * @param {Boolean} hasUserReacted - whether the current user has reacted to this emoji
+
+     */
+    showReactionList(
+        event,
+        reactionListAnchor,
+        users,
+        emojiName,
+        emojiCodes,
+        emojiCount,
+        hasUserReacted,
+    ) {
+        const nativeEvent = event.nativeEvent || {};
+
+        this.reactionListAnchor = reactionListAnchor;
+
+        // Singleton behaviour of ReactionListPopover creates race conditions when user requests multiple ReactionListPopover.
+        // But it is possible that every new request registers new callbacks thus instanceID is used to corelate those callbacks
+        this.instanceID = Math.random().toString(36).substr(2, 5);
```

Comment:
what is the purpose of the substring? We should add a comment if it is necessary

Reply:
the same as above